nautobot_device_lifecycle_mgmt/forms.py

```diff
@@ -65,18 +66,40 @@ def prepare_value(self, value):
         return super().prepare_value(pk_list)
 
 
+class HardwareLCMDynamicModelChoiceField(DynamicModelChoiceMixin, forms.ModelChoiceField):
+    """DynamicModelChoiceField used for 'inventory_item' field in HardwareLCMForm."""
+
+    def to_python(self, value):
+        """Overload 'to_python' in forms.ModelChoiceField to force returning 'part_id' as the field value."""
+        if value in self.empty_values:
+            return None
+        if self.to_field_name == "part_id":
+            return value
+        return super().to_python(value)
+
+
 class HardwareLCMForm(BootstrapMixin, CustomFieldModelFormMixin, RelationshipModelFormMixin):
     """Hardware Device Lifecycle creation/edit form."""
 
-    inventory_item = forms.ModelChoiceField(
-        queryset=InventoryItem.objects.exclude(part_id__exact="")
-        .distinct()
-        .order_by("part_id")
-        .values_list("part_id", flat=True),
+    device_type = DynamicModelChoiceField(queryset=DeviceType.objects.all(), required=False)
+    inventory_item = HardwareLCMDynamicModelChoiceField(
+        queryset=InventoryItem.objects.order_by().distinct("part_id"),
+        query_params={"part_id__nre": "^$", "nautobot_device_lifecycle_mgmt_distinct_part_id": "true"},
         label="Inventory Part ID",
+        display_field="part_id",
         to_field_name="part_id",
+        brief_mode=False,
         required=False,
     )
+    # inventory_item = forms.ModelChoiceField(
+    #     queryset=InventoryItem.objects.exclude(part_id__exact="")
```

Comment:
Should this be left in here?

Reply:
Nope, it should not! I'll get it removed.